launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -338,248 +402,317 @@ void FirstLaunchView::extractGogData()
 	QString fileBin = fileSelection(filterBin);
 	if(fileBin.isEmpty())
 		return;
+
 	QString fileExe = fileSelection(filterExe, QFileInfo(fileBin).absolutePath());
 	if(fileExe.isEmpty())
 		return;
 
-	ui->progressBarGog->setVisible(true);
-	ui->pushButtonGogInstall->setVisible(false);
-	setEnabled(false);
-
 	QTimer::singleShot(100, this, [this, fileBin, fileExe](){ // background to make sure FileDialog is closed...
 		extractGogDataAsync(fileBin, fileExe);
-		ui->progressBarGog->setVisible(false);
-		ui->pushButtonGogInstall->setVisible(true);
 		setEnabled(true);
 		heroesDataUpdate();
 	});
 #endif
 }
 
-void FirstLaunchView::extractGogDataAsync(QString filePathBin, QString filePathExe)
+bool performCopyFlow(const QString &path, FirstLaunchView *self, ProgressOverlay *overlay, bool removeSource = false)
 {
-	logGlobal->info("Extracting gog data from '%s' and '%s'", filePathBin.toStdString(), filePathExe.toStdString());
+    // 1) Scan -> "src \t Target \t Name"
+    overlay->setTitle(QObject::tr("Scanning selected folder..."));
```

Comment:
if you make this function a method, then you won't need to use `QObject::` and when translating this string the context will be clear